### ISSUE TITLE
Feature: WiFi Portal Environments

### ISSUE THREAD

**Issue by quozl** [sugarlabs/sugar#925]
Some locations have a WiFi portal that when accessed using macOS, Windows, Android, or iOS open a browser window and visit a supplied web page for authentication or acceptance of terms and conditions.

Sugar does not provide this feature yet.

**Comment by chimosky** [comment]
I don't think this issue is from sugar as I remember some years back I was able to view wifi portal environments on my XO, I don't have any wifi I could use to test this at the moment though.

**Comment by srevinsaju** [comment]
@chimosky afaik, what @quozl might have mentioned is a `sign-in to connect to WiFi` mode of connection. There are many free public wifi's which requite us to sign in before connection. Android phones show a `WebBrowserIntent` as a notification and ask the user to sign into the Browser to complete connecting to internet. Maybe sugar has not implemented the sign-in to connect method of WiFi,

I hope I did not misinterpret what @quozl  meant. This is what I understood

**Comment by quozl** [comment]
Agreed, it's more than just using Browse to interact with the portal.  Other operating systems launch a minibrowser in response.  We use WebKit for Help in Sugar, so we know we can do that kind of thing, we just need to request and handle the event.

**Comment by chimosky** [comment]
I now understand what you meant as earlier I thought you meant the interaction with the portal using Browse.
